deflect/StreamSendWorker.cpp

```diff
@@ -68,40 +68,43 @@ void StreamSendWorker::_run()
     const auto sendFunc = std::bind(&StreamPrivate::sendPixelStreamSegment,
                                     &_stream, std::placeholders::_1);
 
-    std::unique_lock<std::mutex> lock(_mutex);
     while (true)
     {
+        std::unique_lock<std::mutex> lock(_mutex);
         while (_requests.empty() && _running)
             _condition.wait(lock);
 
         if (!_running)
             break;
 
         const Request& request = _requests.front();
+        const ImageWrapper image(request.image);
+        const uint32_t tasks(request.tasks);
+        PromisePtr promise(request.promise);
```

Comment:
I find the way the request is copied quite confusing. I didn't see at first that "image" was a copy, I thought it was wrapping around "request.image". I would simply do in two lines:

    const auto request = _requests.front();
    _requests.pop_front();

Or if for some reason you prefer having individual variables then I would still suggest C++11 style with auto for readability:

    const auto& request = _requests.front();
    const auto image = request.image;
    const auto tasks = request.tasks;
    const auto promise = request.promise; 
    _requests.pop_front();

Not really needed as those objects are cheap to copy, but it could even be:

    auto& request = _requests.front();
    const auto image = std::move(request.image);
    const auto tasks = std::move(request.tasks);
    const auto promise = std::move(request.promise); 
    _requests.pop_front();